independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateImpl.java

```diff
@@ -231,8 +235,30 @@ protected Engine engine() {
         }
 
         private CompletionStage<String> renderAsyncNoTimeout() {
-            StringBuilder builder = new StringBuilder(1028);
-            return renderData(data(), builder::append).thenApply(v -> builder.toString());
+            StringBuilder builder = new StringBuilder(getCapacity());
+            return renderData(data(), builder::append).thenApply(v -> {
+                String str = builder.toString();
+                capacity.update(str.length());
+                return str;
+            });
+        }
+
+        private int getCapacity() {
+            if (!attributes.isEmpty()) {
+                Object c = getAttribute(TemplateInstance.CAPACITY);
+                if (c != null) {
+                    if (c instanceof Number) {
+                        return ((Number) c).intValue();
+                    } else {
+                        try {
+                            return Integer.parseInt(c.toString());
```

Comment:
I hope this cannot really happen!

Reply:
It can happen if you pass the attribute value as a string, which is possible. We follow this strategy for all other attributes, e.g. you can use `instance.setAttribute("locale", "en")` or `instance.setAttribute("locale", Locale.ENGLISH)` ...